### PR TITLE
[7/n][resources ui] Expose resource job/op dependencies in repo snapshot

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Locates the set of ops & their containing jobs that refer to a particular resource. Packages them up in the snapshot for display in the UI.

## Test Plan

New unit test.

### CODE REVIEW

**Comment by benpankow** [comment]
Current dependencies on/for this PR:
* master
  * **PR #12728** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12728" target="_blank">Graphite</a> 
    * **PR #12729** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12729" target="_blank">Graphite</a> 
      * **PR #12832** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12832" target="_blank">Graphite</a> 
        * **PR #12833** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12833" target="_blank">Graphite</a> 
          * **PR #13007** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13007" target="_blank">Graphite</a> 
            * **PR #13008** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13008" target="_blank">Graphite</a> 
              * **PR #13048** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13048" target="_blank">Graphite</a>  👈
                * **PR #13049** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13049" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13048?utm_source=stack-comment).

File: python_modules/dagster-graphql/dagster_graphql/schema/resources.py [hunk 1]
```diff
@@ -86,6 +94,7 @@ class GrapheneResourceDetails(graphene.ObjectType):
     )
     resourceType = graphene.NonNull(graphene.String)
     assetKeysUsing = graphene.Field(non_null_list(GrapheneAssetKey))
+    jobsOpsUsing = graphene.Field(non_null_list(GrapheneJobWithOps))
```
Comment (dpeng817) [inline python_modules/dagster-graphql/dagster_graphql/schema/resources.py:97]: Any reason we don't just re-use the existing `GrapheneJob` objects here? Just kinda leaves a weird taste in my mouth to have this super bespoke gql schema when we have something else that already has all this info

File: python_modules/dagster/dagster/_core/host_representation/external_data.py [hunk 2]
```diff
@@ -1143,12 +1153,31 @@ def external_repository_data_from_def(
...
 
+    # maps resource key to map of pipeline name to list of solids in that pipeline that use that resource
+    resource_job_usage_map: Dict[str, Dict[str, List[str]]] = defaultdict(lambda: defaultdict(list))
+    for pipeline in pipeline_datas or []:
+        pipeline_name = pipeline.name
+        if is_base_asset_job_name(pipeline_name):
+            continue
+        for solid_name in pipeline.pipeline_snapshot.solid_names:
+            try:
+                solid = pipeline.pipeline_snapshot.get_node_def_snap(solid_name)
```
Comment (dpeng817) [inline python_modules/dagster/dagster/_core/host_representation/external_data.py:1170]: given that node def and job def snaps are basically already retrieved here, makes me even more dubious of creating this new JobWithOps schema

**Review by dpeng817 — COMMENTED** [review]
Would like to hear thoughts regarding schema re-use

File: python_modules/dagster-graphql/dagster_graphql/schema/resources.py (hunk 1 above)
Comment (benpankow) [reply python_modules/dagster-graphql/dagster_graphql/schema/resources.py]: Updated, now returns `GrapheneJob` and `GrapheneSolidHandle`s

**Review by dpeng817 — APPROVED** [review]

**Pull request merged**